src/main/scala/uk/gov/hmrc/zap/api/ZapScan.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.zap.api
+
+import org.scalatest.concurrent.Eventually
+import org.scalatest.time.{Millis, Seconds, Span}
+import play.api.libs.json.Json
+import uk.gov.hmrc.zap.ZapException
+import uk.gov.hmrc.zap.client.ZapClient
+import uk.gov.hmrc.zap.logger.ZapLogger.log
+
+class ZapScan(zapClient: ZapClient) extends Eventually {
+
+  import zapClient._
+  import zapClient.zapConfiguration._
+  import Status._
+
+  implicit override val patienceConfig =
+    PatienceConfig(timeout = scaled(Span(patienceConfigTimeout, Seconds)), interval = scaled(Span(500, Millis)))
+
+  def runAndCheckStatusOfSpider(contextName: String): Unit = {
```

Comment:
For consistency, here the parameter should be the implicit ZapContext

Reply:
This has been changed to use implicit ZapContext